src/game/movement.rs

```diff
@@ -128,7 +128,7 @@ pub fn apply_movement(
         // apply forces and friction
         let mut velocity = movement.velocity + force;
```

Comment:
We should just set this directly by anything that wants to move, rather than applying acceleration * dt to it. 
```suggestion
        let mut velocity = movement.velocity;
```